src/sources.rs

```diff
@@ -89,19 +93,30 @@ impl Iter {
     fn visit_next_directory(&mut self, dir_entry: Entry) -> io::Result<()> {
         let readdir = try!(fs::read_dir(&dir_entry.path));
         self.report.increment("source.visited.directories", 1);
-        let mut children = Vec::<(OsString, bool)>::new();
+        let mut children = Vec::<(OsString, bool, Apath)>::new();
         for entry in readdir {
             let entry = try!(entry);
             let ft = try!(entry.file_type());
-            children.push((entry.file_name(), ft.is_dir()));
+            let mut path = dir_entry.apath.to_string().clone();
+            if path != "/" {
+                path.push('/');
+            }
+            // TODO: Don't be lossy, error if not convertible.
+            path.push_str(&entry.file_name().to_string_lossy());
+
+            if self.excludes.is_match(&path) {
```

Comment:
I'm not sure but would it be reasonable to just match against the apath rather than constructing a separate path? How is it meant to be different?

Reply:
This path is the apath for the entry. If I match the excludes against the apath of the entry then only the directories where excluded. The entry is a `DirType` and because of that it hasn't already an apath. So I just copied the apath creation for the children from the lines below (in origin line 100-114 [sources.rs#L100](https://github.com/sourcefrog/conserve/blob/master/src/sources.rs#L100)) to the actual location...